jwst/master_background/expand_to_2d.py

```diff
@@ -0,0 +1,314 @@
+                             left=0., right=0.)
+
+        background.slits[k].data[:] = bkg_flux.copy()
+
+    return background
+
+
+def bkg_for_image(input, tab_wavelength, tab_background):
+    """Create a 2-D background for an ImageModel.
+
+    Parameters
+    ----------
+    input : JWST ImageModel
+        The input science data.
+
+    tab_wavelength : 1-D ndarray
+        The wavelength column read from the 1-D background table.
+
+    tab_background : 1-D ndarray
+        The flux column read from the 1-D background table, divided by
+        the npixels column.
+
+    Returns
+    -------
+    background : JWST ImageModel
+        A copy of `input` but with the data replaced by the background,
+        "expanded" from 1-D to 2-D.
+    """
+
+    background = input.copy()
+
+    wl_array = get_wavelengths(input)
+    if wl_array is None:
+        raise RuntimeError("Can't determine wavelengths for {}"
+                           .format(type(input)))
+
+    wl_array[np.isnan(wl_array)] = -1.
+
+    # bkg_flux will be a 2-D array, because wl_array is 2-D.
+    bkg_flux = np.interp(wl_array, tab_wavelength, tab_background,
+                         left=0., right=0.)
+
+    background.data[:] = bkg_flux.copy()
+
+    return background
+
+
+def bkg_for_IFUimage(input, tab_wavelength, tab_background):
+    """Create a 2-D background for an IFUImageModel
+
+    Parameters
+    ----------
+    input : JWST IFUImageModel
+        The input science data.
+
+    tab_wavelength : 1-D ndarray
+        The wavelength column read from the 1-D background table.
+
+    tab_background : 1-D ndarray
+        The flux column read from the 1-D background table, divided by
+        the npixels column.
+
+    Returns
+    -------
+    background : JWST IFUImageModel
+        A copy of `input` but with the data replaced by the background,
+        "expanded" from 1-D to 2-D.
+    """
+
+    background = input.copy()
+    background.data[:, :] = 0.
+
+    if input.meta.instrument.name == "NIRSPEC":
+        list_of_wcs = nirspec.nrs_ifu_wcs(input)
+        for ifu_wcs in list_of_wcs:
+
+            xstart = ifu_wcs.bounding_box[0][0]
+            xstop = ifu_wcs.bounding_box[0][1]
+            ystart = ifu_wcs.bounding_box[1][0]
+            ystop = ifu_wcs.bounding_box[1][1]
+
+            # Convert to integers, and add one to the upper limits,
+            # because we want to use these as slice limits.
+            xstart = math.floor(xstart)
+            xstop = math.ceil(xstop) + 1
+            ystart = math.floor(ystart)
+            ystop = math.ceil(ystop) + 1
+            xstart = max(xstart, 0)
+            ystart = max(ystart, 0)
+            xstop = min(xstop, 2048)
+            ystop = min(ystop, 2048)
+
+            shape = (ystop - ystart, xstop - xstart)
+            grid = np.indices(shape, dtype=np.float64)
+            grid[0] += ystart
+            grid[1] += xstart
+            wl_array = ifu_wcs(grid[1], grid[0])[2]
```

Comment:
The block from lines 240-260 can probably be replaced by
```
from gwcs.wcstools import grid_from_bounding_box

        wl_array = grid_from_bounding_box(ifu_wcs.bounding_box)
```
though you should verify it is giving the same result as your code.  This is the standard library function for generating a grid of points from a WCS with a `bounding_box` defined.

Reply:
Just checked it.  Your code and `grid_from_bounding_box` produce identical results for the IFUImage data.